.github/scripts/request_validation.py

```diff
@@ -0,0 +1,77 @@
+import json
+import os
+import sys
+
+import requests
+
+# --------- #
+# Handling arguments
+# --------- #
+dirname = sys.argv[1]
+curl_URL = sys.argv[2]
+extension = ".json"
+
+# --------- #
+# Function definition
+# --------- #
+
+
+def request_validation(data_filepath, 
+                       curl_URL,
+                       headers={'Content-Type': 'application/json'}
+                       ):
+    """
+    Function that, given a fata_filepath (e.g. "path/to/file.json"), a URL (e.g. http://localhost:3020/validate)
+        and the HTTP headers, will do a post request and return the response
+    """
+    with open(data_filepath) as f:
+        data = f.read().replace('\n', '').replace('\r', '').encode('utf-8')
+
+    response = requests.post(url=curl_URL, headers=headers, data=data)
+    return response
+
+
+def get_errors_response(response):
+    """
+    Function that, given a "requests.models.Response" object, will interpret it, asserting that the request
+        was successful and no validation errors (i.e. empty list of errors) were found. If not, it will
+        return the error.
+    """
+    filename = file.name
+    assert (
+            type(response) == requests.models.Response
+        ), "The POST response was not of the correct type"
+    
+    if not response.status_code == 200:
+        error_message = f"The POST response was not successful: instead of 200, the status code was '{response.status_code}' when validating file '{filename}'"
+        return error_message
+
+    # We load the result of the validation
+    val_response_list = json.loads(response.text)
+
+    # If the list is empty "[]", the validation found no errors
+    if not len(val_response_list) == 0:
+        return val_response_list
+
+
+# --------- #
+# Running validations
+# --------- #
+error_dict = {}
+# We iterate over the JSON documents to validate
+for file in os.scandir(dirname):
+    # Check if it's a JSON file
+    if not file.path.endswith(extension):
+        continue
+
+    request = request_validation(data_filepath=file,
+                                 curl_URL=curl_URL)
+
+    val_error = get_errors_response(request)
+    # If there was a validation error, we add it to the dictionary
```

Comment:
Well, personally I prefer one-liners that are easy to interpret for those who have not read the code. Since they don't influence the execution time and can be ignored, or even parsed out, if needed, I tend to have a few in my code. 

Once again, it's personal preference, since I've had to read my own code after a year and I appreciated these. Or would have paid for them to be present in many other scripts. 

In any case, I left only 3 comments that, to my taste, add context.

Reply:
Comments can distract you from reading the code, try to focus more on the readability of the code than putting comments to explain them. I really like it when it feels like you're reading an English text/story when you're reading the code. Using descriptive names and functions can help. You don't have to apply it now only if time permits and yeah just think about it next time when you're writing/refactoring your codes.

for example(may not be the best example but just trying to make a point that you don't need the comments in some parts of your code):

Before:
```
# We iterate over the JSON documents to validate
for file in os.scandir(dirname):
    # Check if it's a JSON file
    if not file.path.endswith(extension):
        continue
```


After:
```
for file in os.scandir(dirname):
    if not is_json(file):
        continue

def is_json(file):
    return file.path.endswith(extension)
```

Reply:
Yep, understood. Will try to keep it in mind, although I tend to be... verbose. Thanks for the feedback, Alegria 🥇